cms/djangoapps/contentstore/management/commands/generate_test_course.py

```diff
@@ -0,0 +1,56 @@
+"""
+Django management command to generate a test course in a specific modulestore
+"""
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand, CommandError
+
+from contentstore.management.commands.utils import user_from_str
+from contentstore.views.course import create_new_course_in_store
+from xmodule.modulestore import ModuleStoreEnum
+
+import json
+import random
+
+
+class Command(BaseCommand):
+    """ Generate a basic course """
+    help = 'Generate a course with settings on studio'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--json',
+            action='store',
+            dest='json',
+            type=str,
+            default=None,
+            help='A json object with the following fields: store, user, name, organization, number, run'
+        )
+
+    def handle(self, *args, **options):
+        # Set default course settings
+        store = "split"
+        user = user_from_str("edx@example.com")
+        name = "test-course"
+        org = "test-course-generator"
+        num = str(random.randint(0, 100))  # Random course number
```

Comment:
I've removed the default case for now, I didn't want to over complicate the command until I got a grip on testing, especially if it isn't so necessary (It might be better to just require a user to specify settings in the config file instead of generating random course numbers)